File: January2022/Java/ValidMountain.java
class Solution {
	// TC : O(n)
	// SC : O(1)
	public boolean validMountainArray(int[] A) {
		int len = A.length;
		if(len==0){
			return false;
		}
		int i = 0;
		for(;i+1<len;i++){
			if(A[i]<A[i+1]){

			} else {
				break;
			}
		}

		if(i==0|| i==len-1){
			return false;
		}

		for(;i+1<len;i++){
			if(A[i]>A[i+1]){

			} else {
				break;
			}
		}
		return i==(len-1);
	}
}

File: January2022/Python/validMountain.py
# Author :  Karthik K (https://www.linkedin.com/in/karthikmunirathinam/)

# TC : O(N)
# SC : O(1)

class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        ind = 0
        l = len(arr)
        while ind+1 < l and arr[ind] < arr[ind+1]:
            ind += 1
        if ind == 0 or ind == l-1:
            return False
        while ind+1 < l and arr[ind] > arr[ind+1]:
            ind += 1
        return ind == l-1
